### DISCUSSION TITLE
EN2-8m WiFi stick

### DISCUSSION THREAD

**Discussion by dorianwoolger** [alienatedsec/solis-ha-modbus-cloud#26] [Q&A] [answered]
Hi 

Please can I confirm that this setup will work with the EN2-8m stick. I can't find mention of it for use with HA.

**Comment by alienatedsec** [comment]
@dorianwoolger It looks like any typical 4-pin data logger. I don't see a reason why it would not work. Which inverter do you have?

**Reply by dorianwoolger** [reply]
Hi. Thanks for the reply. I have the RHI-5K-48E-5G. Installed Sep2021.

**Reply by alienatedsec** [reply]
Those dataloggers work the same way and I guess the naming convention is slightly different nowadays.

**Reply by dorianwoolger** [reply]
Thanks for info this is the info from the datalogger web interface

![image](https://github.com/alienatedsec/solis-ha-modbus-cloud/assets/19670620/bbdc82ff-eb40-4b25-a43b-01b311b3ac1e)

**Reply by alienatedsec** [reply, marked as answer]
Its likely to be a `DLS-W` type. Try Solarman integration https://github.com/StephanJoubert/home_assistant_solarman

It should work without any hardware modifications and report to the Solis Cloud at the same time.

**Reply by dorianwoolger** [reply]
Thanks, that does read all the data fine. Was looking for a solution to control too.

**Reply by alienatedsec** [reply]
Here you go https://github.com/jmccrohan/ha_pyscript_pysolarmanv5

**Reply by dorianwoolger** [reply]
Thanks again. Will take a look. Will that play nicely alongside standard Solis app or is the solution using the waveshare devices a better way to go?

In your opinion, which will govern the most reliable setup with HA and SolisCloud both working side by side. 

Many thanks for your time so far with the advice.

**Reply by alienatedsec** [reply]
@dorianwoolger I had the same type of datalogger. Not sure if it was 100% faulty but it wasnt performing well when queried with additional to SolisCloud requests. Check my diagram how I used it in the past. 

If it works with the Solarman integration and the additional scripts from the repo by @jmccrohan then have a try to save time and money on more hardware and soldering. 

Otherwise, my design should work perfectly and SolaX integration does everything out of box.

**Reply by dorianwoolger** [reply]
@alienatedsec Always nice to save money but the main aim is to have a solid system that works with HA and SolisCloud.  On the basis that HA along with the plug-ins so far have cost nothing, a few quid on waveshare kit and a switch is quite justified.

I have no issue with soldering and quite enjoy that part. 

Solarman seems to be working well and and so far not interruptions to SolisCloud either.

I have had a read of the documents and think I will go with your solution documented in this Git repo and it seems like the SolaX integration will be much easier to manage.

Thank you very much for your replies.

**Comment by ilovemypvc** [comment]
I need help with the wifi logger , the green connector's wire broke off and i don't know which colour corresponds to which number 
![17466834557868321829605131639566](https://github.com/user-attachments/assets/3e47e4a4-a48e-45d9-9dee-d83d62a7fb74)

**Reply by alienatedsec** [reply]
You could determine VCC and GND with a multimeter. The other two by trial and error. The power pins are on my diagram.

**Reply by ilovemypvc** [reply]
Where can i find the diagram , btw I only know the basics of electronics I'm learning and thanks for the reply

**Reply by alienatedsec** [reply]
Can I have pictures of both sides of the board near the cable connector?

**Reply by alienatedsec** [reply]
Sorted it? @ilovemypvc

**Reply by ilovemypvc** [reply]
![PXL_20250509_124914818.MP.jpg](https://github.com/user-attachments/assets/05f2d711-0101-4092-8fc4-2e2d6c26a9f1)

![PXL_20250509_124921598.MP.jpg](https://github.com/user-attachments/assets/1f7a0dbf-c8cc-4273-aad6-0518b312cc78)

**Reply by alienatedsec** [reply]
@ilovemypvc On the Socket and the PLUG, there are PIN numbers next to each hole.

1 - RED
2 - BLACK
3 - YELLOW
4 - BLUE

**Reply by ilovemypvc** [reply]
Yo thank you so much for the help , ill sotter it then ill check
